app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -823,6 +826,11 @@ class BraveVPNService : VpnService(), ConnectionCapabilityMonitor.NetworkListene
             Log.i(LOG_TAG, "$FILE_LOG_TAG ORBOT proxy change - restart vpn ${appMode.getProxyMode()}")
             restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
         }
+
+        if(PersistentState.NETWORK == key){
+            connectionMonitor?.removeCallBack()
+            connectionMonitor = ConnectionMonitor(this, this)
+        }
```

Comment:
Don't recreate the object, instead forward the changes to the preference to connection-monitor.

Reply:
Modified the implementation. Now the preference change will invoke the onPreferenceChange()